### PR TITLE
feat(community): Extend DocxLoader to load .doc files

### PR DESCRIPTION
**Pull request by Fibii**
Extends the existing `DocxLoader` to handle loading `.doc` files.
Uses [word-extractor](https://www.npmjs.com/package/word-extractor) as a peer dependency for loading .doc

### CODE REVIEW

File: docs/core_docs/docs/integrations/document_loaders/file_loaders/docx.mdx [hunk 1]
```diff
@@ -23,3 +23,28 @@ const loader = new DocxLoader(
...
+import { DocxLoader } from "@langchain/community/document_loaders/fs/docx";
+
+const loader = new DocxLoader(
+  "src/document_loaders/tests/example_data/attentio.docx",
+  {
+    type: "doc",
+  }
+);
+
+const docs = await loader.load();
```
Comment (jacoblee93) [inline docs/core_docs/docs/integrations/document_loaders/file_loaders/docx.mdx:50]: Missing closing markdown fence

File: docs/core_docs/docs/integrations/document_loaders/file_loaders/docx.mdx [hunk 2]
```diff
@@ -23,3 +23,28 @@ const loader = new DocxLoader(
 
 const docs = await loader.load();
 ```
+
+
+# Doc files
+
+This example goes over how to load data from doc files.
```
Comment (jacoblee93) [inline docs/core_docs/docs/integrations/document_loaders/file_loaders/docx.mdx:30]: I would also mention you need another dep in the `Setup` section

File: libs/langchain-community/src/document_loaders/fs/docx.ts [hunk 3]
```diff
@@ -53,3 +119,15 @@ async function DocxLoaderImports() {
...
 }
+
+async function DocLoaderImports() {
+  try {
+    const WordExtractor = await import("word-extractor");
+    return WordExtractor.default;
+  } catch (e) {
+    console.error(e);
+    throw new Error(
+      "Failed to load word-extractor. Please install it with eg. `npm install word-extractor`."
```
Comment (jacoblee93) [inline libs/langchain-community/src/document_loaders/fs/docx.ts:130]: No way to just use `mammoth` is there?

**Review by jacoblee93 — COMMENTED** [review]
Thank you! Some small nits and questions

Can you please also run `yarn format` from root? The `.mdx` files get formatted as well

File: libs/langchain-community/src/document_loaders/fs/docx.ts (hunk 3 above)
Comment (Fibii) [reply libs/langchain-community/src/document_loaders/fs/docx.ts:130]: [No](https://github.com/mwilliamson/mammoth.js/issues/2#issuecomment-23871676), `mammoth` can only handle `.docx`

**Comment by Fibii** [comment]
> Thank you! Some small nits and questions
> 
> Can you please also run `yarn format` from root? The `.mdx` files get formatted as well

Done

**Review by jacoblee93 — APPROVED** [review]
Thanks!

**Pull request merged**